Allows to mutate `const` with non-null assertion

**TypeScript Version:**  2.8.0-dev.20180209

**Code**

```ts
declare const x: number | undefined;
x!++;
```

**Expected behavior:**

```
src/a.ts(2,2): error TS2540: Cannot assign to 'x' because it is a constant or a read-only property.
```

**Actual behavior:**

No error.

--- a/src/compiler/utilities.ts
+++ b/src/compiler/utilities.ts
@@ -1754,6 +1754,7 @@ namespace ts {
                 case SyntaxKind.ParenthesizedExpression:
                 case SyntaxKind.ArrayLiteralExpression:
                 case SyntaxKind.SpreadElement:
+                case SyntaxKind.NonNullExpression:
                     node = parent;
                     break;
                 case SyntaxKind.ShorthandPropertyAssignment:

--- a/tests/baselines/reference/constWithNonNull.errors.txt
+++ b/tests/baselines/reference/constWithNonNull.errors.txt
@@ -1,0 +1,11 @@
+tests/cases/compiler/constWithNonNull.ts(4,1): error TS2540: Cannot assign to 'x' because it is a constant or a read-only property.
+
+
+==== tests/cases/compiler/constWithNonNull.ts (1 errors) ====
+    // Fixes #21848
+    
+    declare const x: number | undefined;
+    x!++;
+    ~
+!!! error TS2540: Cannot assign to 'x' because it is a constant or a read-only property.
+    

--- a/tests/baselines/reference/constWithNonNull.js
+++ b/tests/baselines/reference/constWithNonNull.js
@@ -1,0 +1,10 @@
+//// [constWithNonNull.ts]
+// Fixes #21848
+
+declare const x: number | undefined;
+x!++;
+
+
+//// [constWithNonNull.js]
+// Fixes #21848
+x++;

--- a/tests/baselines/reference/constWithNonNull.symbols
+++ b/tests/baselines/reference/constWithNonNull.symbols
@@ -1,0 +1,9 @@
+=== tests/cases/compiler/constWithNonNull.ts ===
+// Fixes #21848
+
+declare const x: number | undefined;
+>x : Symbol(x, Decl(constWithNonNull.ts, 2, 13))
+
+x!++;
+>x : Symbol(x, Decl(constWithNonNull.ts, 2, 13))
+

--- a/tests/baselines/reference/constWithNonNull.types
+++ b/tests/baselines/reference/constWithNonNull.types
@@ -1,0 +1,11 @@
+=== tests/cases/compiler/constWithNonNull.ts ===
+// Fixes #21848
+
+declare const x: number | undefined;
+>x : number
+
+x!++;
+>x!++ : number
+>x! : any
+>x : any
+

--- a/tests/cases/compiler/constWithNonNull.ts
+++ b/tests/cases/compiler/constWithNonNull.ts
@@ -1,0 +1,4 @@
+// Fixes #21848
+
+declare const x: number | undefined;
+x!++;

